packages/datadog-plugin-http2/test/server.spec.js

```diff
@@ -0,0 +1,259 @@
+  url = new URL(url)
+  return new Promise((resolve, reject) => {
+    const client = http2
+      .connect(url.origin)
+      .on('error', reject)
+
+    const req = client.request({
+      ':path': url.pathname,
+      ':method': 'GET'
+    })
+    req.on('error', reject)
+
+    if (signal) {
+      signal.on('abort', () => req.destroy())
+    }
+
+    const chunks = []
+    req.on('data', (chunk) => {
+      chunks.push(chunk)
+    })
+    req.on('end', () => {
+      resolve(Buffer.concat(chunks))
+    })
+
+    req.end()
+  })
+}
+
+describe('Plugin', () => {
+  let http2
+  let listener
+  let appListener
+  let tracer
+  let port
+  let app
+
+  describe('http2/server', () => {
+    beforeEach(() => {
+      tracer = require('../../dd-trace')
+      listener = (req, res) => {
+        app && app(req, res)
+        res.writeHead(200)
+        res.end()
+      }
+    })
+
+    beforeEach(() => {
+      return getPort().then(newPort => {
+        port = newPort
+      })
+    })
+
+    afterEach(() => {
+      appListener && appListener.close()
+      app = null
+      return agent.close({ ritmReset: false })
+    })
+
+    describe('canceled request', () => {
+      beforeEach(() => {
+        listener = (req, res) => {
+          setTimeout(() => {
+            app && app(req, res)
+            res.writeHead(200)
+            res.end()
+          }, 500)
+        }
+      })
+
+      beforeEach(() => {
+        return agent.load('http2')
+          .then(() => {
+            http2 = require('http2')
+          })
+      })
+
+      beforeEach(done => {
+        const server = http2.createServer(listener)
+        appListener = server
+          .listen(port, 'localhost', () => done())
+      })
+
+      it('should send traces to agent', (done) => {
+        app = sinon.stub()
+        agent
+          .use(traces => {
+            expect(app).not.to.have.been.called // request should be cancelled before call to app
+            expect(traces[0][0]).to.have.property('name', 'web.request')
+            expect(traces[0][0]).to.have.property('service', 'test')
+            expect(traces[0][0]).to.have.property('type', 'web')
+            expect(traces[0][0]).to.have.property('resource', 'GET')
+            expect(traces[0][0].meta).to.have.property('span.kind', 'server')
+            expect(traces[0][0].meta).to.have.property('http.url', `http://localhost:${port}/user`)
+            expect(traces[0][0].meta).to.have.property('http.method', 'GET')
+            expect(traces[0][0].meta).to.have.property('http.status_code', '200')
+          })
+          .then(done)
+          .catch(done)
+
+        // Don't use real AbortController because it requires 15.x+
+        const ac = new MockAbortController()
+        request(http2, `http://localhost:${port}/user`, {
+          signal: ac.signal
+        })
+        setTimeout(() => { ac.abort() }, 100)
+      })
+    })
+
+    describe('without configuration', () => {
```

Comment:
Most of this test is pretty much the same as `http`. Can some parts (or the entire suite) be easily reused? If not in its current state, let's add a TODO so we can merge some of it later.

Reply:
That will just become a mess if we restore parts of the support for non-compatibility mode stuff later. The http and http2 modules have a lot of very different behaviour so I think it's probably best to just keep the tests fully separate.